lib/azure/armrest/resource_group_service.rb

```diff
@@ -0,0 +1,90 @@
+module Azure
+  module Armrest
+    class ResourceGroupService < ArmrestService
+      # The provider used in http requests. The default is 'Microsoft.Resources'
+      attr_reader :provider
+
+      # Creates and returns a new ResourceGroupService object.
+      #
+      def initialize(_armrest_configuration, options = {})
+        super
+
+        @provider = options[:provider] || 'Microsoft.Resources'
+
+        set_service_api_version(options, 'resourceGroups')
+      end
+
+      # List all the resources for the current subscription. You can optionally
+      # pass :top or :filter options as well to restrict returned results.
+      #
+      # If you pass a :resource_group option, then only resources for that
+      # resource group are returned.
+      #
+      # Examples:
+      #
+      #   rgs = ResourceGroupService.new
+      #   rgs.list(:top => 2)
+      #   rgs.list(:filter => "location eq 'centralus'")
+      #
+      def list(options = {})
+        url = File.join(Azure::Armrest::COMMON_URI, subscription_id, 'resourcegroups')
+
+        url << "?api-version=#{api_version}"
+        url << "&$top=#{options[:top]}" if options[:top]
+        url << "&$filter=#{options[:filter]}" if options[:filter]
+
+        response = rest_get(URI.escape(url))
+
+        JSON.parse(response.body)["value"]
+      end
+
+      # Creates a new +group+ in +location+ for the current subscription.
+      # You may optionally apply +tags+.
+      #
+      def create_resource_group(group, location, tags = nil)
```

Comment:
we should standardize methods in all `service` classes with `list`, `get`, `create`, `delete`, `update`. This method becomes `create`